File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int size_nums = nums.size();
        std::vector<int> result (size_nums * 2);
        for (int i = 0; i < size_nums; i++) {
            result[i] = nums[i];
            result[i + size_nums] = nums[i];
        }

        return result;
    }
};

File: Data Structures & Algorithms/combination-target-sum-ii/submission-0.cs
public class Solution {
    public List<List<int>> CombinationSum2(int[] candidates, int target) {
        List<List<int>> result = new List<List<int>>();
        List<int> subset = new List<int>();
        Array.Sort(candidates);
        Dfs(candidates, 0, 0, target, result, subset);
        return result;
    }

    private void Dfs(int[] nums, int index, int sum, int target, List<List<int>> result, List<int> subset) {
        if (sum == target) {
            result.Add(new List<int> (subset));
            return;
        }

        if (index >= nums.Length || sum > target) return;
        
        subset.Add(nums[index]);
        Dfs(nums, index + 1, sum + nums[index], target, result, subset);

        subset.Remove(nums[index]);
        int next = index + 1;
        while (next < nums.Length && nums[index] == nums[next]) next++;
        Dfs(nums, next, sum, target, result, subset);
    }
}
